machine/v1/types_nutanixprovider.go

```diff
@@ -0,0 +1,244 @@
+}
+
+// NutanixMachineProviderStatus is the type that will be embedded in a Machine.Status.ProviderStatus field.
+// It contains nutanix-specific status information.
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type NutanixMachineProviderStatus struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// ready is true when the provider resource is ready.
+	// +optional
+	Ready bool `json:"ready"`
+
+	// vmUUID is the VM's UUID
+	// +optional
+	VmUUID *string `json:"vmUUID,omitempty"`
+
+	// conditions is a set of conditions associated with the Machine to indicate
+	// errors or other status
+	// +optional
+	Conditions []NutanixMachineProviderCondition `json:"conditions,omitempty"`
+}
+
+// NutanixMachineProviderConditionType is a valid value for NutanixMachineProviderCondition.Type
+type NutanixMachineProviderConditionType string
+
+// Valid conditions for an Nutanix VM instance.
+const (
+	// NutanixMachineCreation indicates whether the machine's VM has been created or not. If not,
+	// it should include a reason and message for the failure.
+	NutanixMachineCreation NutanixMachineProviderConditionType = "MachineCreation"
+
+	// NutanixMachineUpdate indicates whether the machine's VM has been updated or not. If not,
+	// it should include a reason and message for the failure.
+	NutanixMachineUpdate NutanixMachineProviderConditionType = "MachineUpdate"
+
+	// NutanixMachineDeletion indicates whether the machine's VM has been deleted or not. If not,
+	// it should include a reason and message for the failure.
+	NutanixMachineDeletion NutanixMachineProviderConditionType = "MachineDeletion"
+)
+
+// NutanixMachineProviderConditionReason is reason for the condition's last transition.
+type NutanixMachineProviderConditionReason string
+
+const (
+	// NutanixMachineCreationSucceeded indicates machine creation success.
+	NutanixMachineCreationSucceeded NutanixMachineProviderConditionReason = "MachineCreationSucceeded"
+	// NutanixMachineCreationFailed indicates machine creation failure.
+	NutanixMachineCreationFailed NutanixMachineProviderConditionReason = "MachineCreationFailed"
+	// NutanixMachineCreationSucceeded indicates machine update success.
+	NutanixMachineUpdateSucceeded NutanixMachineProviderConditionReason = "MachineUpdateSucceeded"
+	// NutanixMachineCreationFailed indicates machine update failure.
+	NutanixMachineUpdateFailed NutanixMachineProviderConditionReason = "MachineUpdateFailed"
+	// NutanixMachineCreationSucceeded indicates machine deletion success.
+	NutanixMachineDeletionSucceeded NutanixMachineProviderConditionReason = "MachineDeletionSucceeded"
+	// NutanixMachineCreationFailed indicates machine deletion failure.
+	NutanixMachineDeletionFailed NutanixMachineProviderConditionReason = "MachineDeletionFailed"
+)
```

Comment:
We already have a number of common condition types in the v1beta1 package, it may be better to keep generic condition types and reasons rather than creating specific Nutanix ones

Reply:
I raised this question at the slack channel, see https://coreos.slack.com/archives/C0211848DBN/p1644522435725449. Since currently these common condition types in the v1beta1 package, it's not good to cross-reference those from the v1 package. We have to use the "Nutanix"-prefixed condition types and constants here, unless RH move those common condition types from  the "machine/v1beta1" package to a none version-specific package, or duplicate those in the "machine/v1" package.

Reply:
Moved these constants to the mapi-nutanix repo. And use the upstream metav1.Condition type.